### PR TITLE
Updating contributing.md for the monorepo

### PR DESCRIPTION
**Pull request by xomiamoore**
## Description
Updating contributing.md to reflect the new monorepo structure.

I especially need guidance on the `Building Camunda from Source` section as I am not sure how that has changed since the architecture streamlining.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -133,7 +75,7 @@ follow the following steps:
    git checkout -b 123-adding-bpel-support`
```
Comment (lzgabel) [inline CONTRIBUTING.md:75]: @xomiamoore 
❓  remove `` ` ``

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,126 +1,68 @@
...
+## Prerequisites
+### Contributor License Agreement
+
+You will be asked to sign our [Contributor License Agreement](https://cla-assistant.io/camunda-community-hub/community) when you open a Pull Request. We are not
+asking you to assign copyright to us but to give us the right to distribute
+your code without restriction. We ask this of all contributors to
+assure our users of the origin and continuing existence of the code. 
 
-### Build profiling
+*Note: In most cases, you will only need to sign the CLA once.*
```
Comment (korthout) [inline CONTRIBUTING.md:35]: 🔧 Pleae use one of [GitHub's markdown alerts](https://github.com/orgs/community/discussions/16925) for notes

```md
> [!NOTE]  
> In most cases, you will only need to sign the CLA once.
```

> [!NOTE]  
> In most cases, you will only need to sign the CLA once.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,126 +1,68 @@
...
 
-The development team continues to push for a performant build.
-To investigate where the time is spent, you can run your maven command with the `-Dprofile` option.
-This will generate a profiler report in the `target` folder.
+### Code of Conduct
 
-## Report issues or contact developers
+This project adheres to the [Camunda Code of Conduct](https://camunda.com/events/code-conduct/).
+By participating, you are expected to uphold this code. Please [report](https://camunda.com/events/code-conduct/reporting-violations/)
+unacceptable behavior as soon as possible.
```
Comment (korthout) [inline CONTRIBUTING.md:41]: 🔧 I recommend wrapping lines (i.e. starting new lines) for new sentences only. This nudges authors to rewrite sentences to be shorter which typically leads to simpler text that is easier to read. It also makes it easier to review change in git because it highlights changes by removing and adding individual lines.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,126 +1,68 @@
...
+### Starting on an issue
 
-The `main` branch contains the current in-development state of the project. To work on an issue,
-follow the following steps:
+The `main` branch contains the current in-development state of the project. To work on an issue, follow these steps:
 
 1. Check that a [GitHub issue][issues] exists for the task you want to work on.
    If one does not, create one. Refer to the [issue guidelines](#github-issue-guidelines).
-2. Check that no one is already working on the issue, and make sure the team would accept a pull request for this topic. Some topics are complex and may touch multiple of [Camunda's Components](https://docs.camunda.io/docs/components/), requiring internal coordination.
+2. Check that no one is already working on the issue, and make sure the team would accept a pull request for this topic. Some topics are complex and may touch multiple of [Camunda's cmponents](https://docs.camunda.io/docs/components/), requiring internal coordination.
```
Comment (korthout) [inline CONTRIBUTING.md:65]: 🔧 Typo

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -301,37 +243,83 @@ This is optional but encouraged.
...
+## Build Camunda from source
 
-## Licenses
+Camunda is a multi-module Maven project. To **quickly** build all components,
+run the command: `mvn clean install -Dquickly` in the root folder.
 
-Zeebe, Operate, and Tasklist source files are made available under the
-[Camunda License Version 1.0](/licenses/CAMUNDA-LICENSE-1.0.txt) except for the parts listed
-below, which are made available under the [Apache License, Version
-2.0](/licenses/APACHE-2.0.txt).  See individual source files for details.
```
Comment (korthout) [inline CONTRIBUTING.md:317]: ❓ Why was this deleted?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -301,37 +243,83 @@ This is optional but encouraged.
...
+> [!NOTE]
+> All Java modules in Zeebe are built and tested with JDK 21. Most modules use language level
+> 21, exceptions are: zeebe-bpmn-model, zeebe-client-java, zeebe-gateway-protocol,
+> zeebe-gateway-protocol-impl, zeebe-protocol and zeebe-protocol-jackson which use language level 8
```
Comment (korthout) [inline CONTRIBUTING.md:255]: ❓ Should we also document the JDK requirements for Operate/Tasklist/etc?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -301,37 +243,83 @@ This is optional but encouraged.
...
-- Gateway Protocol Implementation ([gateway-protocol-impl](/gateway-protocol-impl))
-- BPMN Model API ([bpmn-model](/bpmn-model))
+For contributions to Camunda, building quickly is typically sufficient.
+However, users of Camunda are recommended to build the full distribution.
 
-If you would like to contribute something, or simply want to hack on the code
-this document should help you get started.
+To fully build the Zeebe distribution, run the command: `mvn clean install -DskipTests` in the root folder.
+This is slightly slower than building quickly but ensures the distribution is assembled completely.
+The resulting Zeebe distribution can be found in the folder `dist/target`, i.e.
```
Comment (korthout) [inline CONTRIBUTING.md:268]: 🔧 This should probably be adjusted for Camunda's mono repo

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -301,37 +243,83 @@ This is optional but encouraged.
...
+- `gateway-protocol` contains the gRPC definitions for the Zeebe client-to-gateway protocol
+- `zb-db` is our RocksDB wrapper for state management
+- `engine`  is the implementation of the event stream processor
+- `broker` contains the Zeebe broker which is the server side of Zeebe
+- `client-java` contains the Java Zeebe client
+- `atomix` contains transport, membership, and consensus algorithms
+- `benchmark` contains utilities the team uses to run load tests
+- `exporters/elasticsearch-exporter` contains the official Elasticsearch exporter for Zeebe
+- `journal` contains the append-only log used by the consensus algorithm
+- `snapshots` abstracts how state snapshots (i.e. `zb-db`) are handled
```
Comment (korthout) [inline CONTRIBUTING.md:304]: 🔧 Some of these modules have been moved into the `zeebe` subfolder. I suggest we adjust this list of modules to explain the root-level folders, and move the Zeebe module specific ones into a README in the zeebe subfolder

**Review by korthout — CHANGES_REQUESTED** [review]
Thanks @xomiamoore 

❌ Please follow our [git commit message guidelines](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#commit-message-guidelines)

🔧 Please split up changes into smaller commits to make reviews easier. It's currently hard to notice that some parts were removed while others were moved. I wonder whether the deleted parts should be removed at all. I may have missed other deleted parts.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,126 +1,68 @@
...
-#### Test Troubleshooting
-
-- If you encounter issues (like `java.lang.UnsatisfiedLinkError: failed to load the required native library`) while running the test StandaloneGatewaySecurityTest.shouldStartWithTlsEnabled take a look at https://github.com/camunda/camunda/issues/10488 to resolve it.
+# Contributing to Camunda
+
+We welcome new contributions. We take pride in maintaining and encouraging a friendly, welcoming, and collaborative community.
+
+Anyone is welcome to contribute to Camunda! The best way to get started is to choose an existing [issue](#starting-on-an-issue).
+
+Core code contributions must fit into the scope of Camunda products. The ultimate decision is up to the core Camunda team. If your vision does not fit in the core Camunda product, you may be directed to the [Camunda Community Hub](https://github.com/camunda-community-hub) and/or [Camunda Marketplace](https://marketplace.camunda.com/en-US/home) instead.
```
Comment (MaxTru) [inline CONTRIBUTING.md:7]: Do we really feel this is necessary? It reads a bit deterring to me. It is kind of clear, that Camunda is the maintainer of this repo and will either accept or reject PRs. 

Maybe a simple note (not that prominent) that there is also a community hub would do it as well.

File: CONTRIBUTING.md (hunk 6 above)
Comment (MaxTru) [reply CONTRIBUTING.md:255]: @korthout I'd assume that we now be design have the same JDK for the core modules being Zeebe, Operate, Tasklist (and Optimize in the near future). Hence I propose to just state 

```suggestion
> All Camunda core modules are built and tested with JDK 21. Most modules use language level
> 21, exceptions are: zeebe-bpmn-model, zeebe-client-java, zeebe-gateway-protocol,
> zeebe-gateway-protocol-impl, zeebe-protocol and zeebe-protocol-jackson which use language level 8
```

File: CONTRIBUTING.md (hunk 7 above)
Comment (MaxTru) [reply CONTRIBUTING.md:268]: 👍 
@xomiamoore maybe you go through the build process yourself, and adjust this accordingly?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -301,37 +243,83 @@ This is optional but encouraged.
...
+```
 
-This project adheres to the [Camunda Code of Conduct](https://camunda.com/events/code-conduct/).
-By participating, you are expected to uphold this code. Please [report](https://camunda.com/events/code-conduct/reporting-violations/)
-unacceptable behavior as soon as possible.
+This distribution can be containerized with Docker (i.e. build a Docker image) by running:
+
+```
+docker build \
+  --tag camunda/zeebe:local \
```
Comment (MaxTru) [inline CONTRIBUTING.md:279]: Should be adjusted in line with monoRepo changes.

File: CONTRIBUTING.md (hunk 8 above)
Comment (MaxTru) [reply CONTRIBUTING.md:304]: 👍

**Review by MaxTru — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (xomiamoore) [reply CONTRIBUTING.md:317]: @korthout -- this is also listed in the README, so I thought it'd be easier to keep it up to date in one place vs. two. Open to changing that if you think it should also be listed here.

File: CONTRIBUTING.md (hunk 9 above)
Comment (xomiamoore) [reply CONTRIBUTING.md:7]: Makes sense, I adjusted this section.

File: CONTRIBUTING.md (hunk 5 above)
Comment (xomiamoore) [reply CONTRIBUTING.md:317]: @korthout, this is also listed in the README, so I thought it'd be easier to keep it up to date in one place vs. two. Open to changing that if you think the licenses should also be listed here.

File: CONTRIBUTING.md (hunk 8 above)
Comment (xomiamoore) [reply CONTRIBUTING.md:304]: Moved the Zeebe-specific folders to a Zeebe README, and added what I could to this section. I'm not sure what some folders are or if they need to be explained.

File: CONTRIBUTING.md (hunk 3 above)
Comment (xomiamoore) [reply CONTRIBUTING.md]: I think I fixed the text wrapping in all the spots where it was, but let me know if you spot any other places.

File: CONTRIBUTING.md (hunk 7 above)
Comment (xomiamoore) [reply CONTRIBUTING.md]: @MaxTru @korthout, Are the steps the same as these, or different? I need a bit more context to get started here.

**Comment by xomiamoore** [comment]
@korthout, @MaxTru, made some updates that are ready for your feedback. I also need a bit more info on building Camunda from source before I can get started on that part.

As a heads up, I'm out next week and the week after is the retreat, so that's when I'll be able to resolve any other feedback. Thank you!

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,122 +1,58 @@
-# Contributing to Zeebe
-
-* [Build Zeebe from source](#build-zeebe-from-source)
```
Comment (MaxTru) [inline CONTRIBUTING.md:3]: I would propose to keep "build from source" at one of the first steps / at the top of the page, since this a very foundation to contribute.

**Review by MaxTru — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 10 above)
Comment (MaxTru) [reply CONTRIBUTING.md:225]: @xomiamoore this is still to be adjusted I think?

**Comment by xomiamoore** [comment]
[Update Contributing.md for Camunda monorepo](https://trello.com/c/BkkzYZKl/3196-update-contributingmd-for-camunda-monorepo)

File: CONTRIBUTING.md (hunk 11 above)
Comment (xomiamoore) [reply CONTRIBUTING.md:3]: @MaxTru, I moved it up higher, but not right at the top. My thinking is that the doc is in order -- first prerequisites for everyone (CLA, CoC) to more specific. Since some may be looking for our Issue guidelines and you don't need to build from source for that (correct me if I'm wrong)-- I put those higher. But let me know if you feel strongly that the build process should be higher up!

**Comment by xomiamoore** [comment]
@MaxTru @korthout Hey there, @nloding and I tested the build instructions. They work but only for Zeebe, and it looks like with the architecture streamlining, the unified build will not be available til 8.7, so each component has separate instructions for now.

I added more context and links to each component's README for their build guide, but Identity and Optimize don't have instructions currently, so those will need to be added. 

Looking forward to your feedback!

**Comment by MaxTru** [comment]
> @MaxTru @korthout Hey there, @nloding and I tested the build instructions. They work but only for Zeebe, and it looks like with the architecture streamlining, the unified build will not be available til 8.7, so each component has separate instructions for now.


Hi @xomiamoore ,
I think there is a misunderstanding here. I also just tried it locally.

Put:
```
mvn clean install -DskipTests
```

This will build
```
camunda-zeebe-8.6.0-SNAPSHOT.tar.gz
```

If unpacked, it will contain amongst other:
```
-rwxr-xr-x  1 maxtru  staff  3755 19 Jul 09:44 broker
-rw-r--r--  1 maxtru  staff  3243 19 Jul 09:44 broker.bat
-rwxr-xr-x  1 maxtru  staff  3757 19 Jul 09:44 camunda
-rw-r--r--  1 maxtru  staff  3245 19 Jul 09:44 camunda.bat
-rwxr-xr-x  1 maxtru  staff  3757 19 Jul 09:44 gateway
-rw-r--r--  1 maxtru  staff  3245 19 Jul 09:44 gateway.bat
-rwxr-xr-x  1 maxtru  staff  3768 19 Jul 09:44 migrate
-rw-r--r--  1 maxtru  staff  3256 19 Jul 09:44 migrate.bat
-rwxr-xr-x  1 maxtru  staff  3757 19 Jul 09:44 operate
-rw-r--r--  1 maxtru  staff  3245 19 Jul 09:44 operate.bat
-rwxr-xr-x  1 maxtru  staff  3752 19 Jul 09:44 restore
-rw-r--r--  1 maxtru  staff  3240 19 Jul 09:44 restore.bat
-rwxr-xr-x  1 maxtru  staff  3759 19 Jul 09:44 tasklist
-rwxr-xr-x  1 maxtru  staff  3778 19 Jul 09:44 tasklist-migrate
-rw-r--r--  1 maxtru  staff  3266 19 Jul 09:44 tasklist-migrate.bat
-rw-r--r--  1 maxtru  staff  3247 19 Jul 09:44 tasklist.bat
```

Now just double click "camunda" to start all camunda - incl. Operate and Tasklist. just double click "operate" to start e.g., only operate. And so on...

Can you please rework the PR based on this?

**Comment by nloding** [comment]
@MaxTru - at this point, we need more guidance from the engineering team(s). There is no documentation about how to build and run this new distribution for our community, and you're asking two people who have never built Camunda/Zeebe from source to write it from scratch with no detail to go on other than the current documentation being wrong or incomplete.

When I ran the commands in the Dockerfile and the existing CONTRIBUTING.md, I did not get a .tar.gz file (screenshot), and I do not know why that might be; I only get the Zeebe JAR, which is 229Kb. There are, of course, some large changes in the Dockerfiles and scripts available between the 8.6.0-alpha3 tag I was testing with and the main branch, so I don't know where to look for correct information. If someone can answer these questions and provide the relevant build commands/parameters, I can help @xomiamoore fill in the rest of the details:

- Is the `-Dquickly` switch expected to build the full FatJAR in the future? If so, is there an ETA?
- If yes to above, is `-Dquickly` still "typically sufficient" for contributions?
- If no, will the only way to build Camunda from source going forward be the longer build command? If so, which command is that? (`mvn clean install -DskipTests -DskipChecks` and/or `mvn -B -am -pl dist package -T1C -DskipChecks -DskipTests` [this was from the camunda.Dockerfile]).
- How do the "StandaloneOperate" etc. applications work? Do you need to run Zeebe also? Do you need a local Postgres or Elasticsearch instance? If yes, how do you configure those when building/running locally? Are there any other external dependencies needed beyond Postgres and Elastic?
- Related to the previous question, if there are dependencies between the applications, is there a startup order? For instance, start the brokers, then the gateway, then the application you want to test?
- Are any of these steps planned to be supported in a Docker container (e.g., the camunda.Dockerfile image)?

image

**Comment by MaxTru** [comment]
@nloding @xomiamoore sorry for the late response, just saw this in my mails today.

> Is the -Dquickly switch expected to build the full FatJAR in the future? If so, is there an ETA?
> If yes to above, is -Dquickly still "typically sufficient" for contributions?
> If no, will the only way to build Camunda from source going forward be the longer build command? If so, which command is that? (mvn clean install -DskipTests -DskipChecks and/or mvn -B -am -pl dist package -T1C -DskipChecks -DskipTests [this was from the camunda.Dockerfile]).

There is no change with the `quickly` flag compared to the status-quo. What the the current [readme](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#build-zeebe-from-source) states is still correct:
> Zeebe is a multi-module maven project. To quickly build all components, run the command: mvn clean install -Dquickly in the root folder.
> ...
> To fully build the Zeebe distribution, run the command: mvn clean install -DskipTests in the root folder. This is slightly slower than building quickly but ensures the distribution is assembled completely. The resulting Zeebe distribution can be found in the folder dist/target, i.e.

=> with `quickly` tests will be skipped and **no distribution will be assembled**. This means: as a developer you can use this to quickly build camunda (e.g., validate that it builds). If you want to try the full distribution end2end (e.g., start it as a user would start it), you must not use the quickly flag. Same is true for the skipTests flag: if you just want to try locally really quick, then skipTests. If you want to be sure that everything works, I would not skipTests. 

Does this help? 

> How do the "StandaloneOperate" etc. applications work? Do you need to run Zeebe also? Do you need a local Postgres or Elasticsearch instance? If yes, how do you configure those when building/running locally? Are there any other external dependencies needed beyond Postgres and Elastic?

* I personally would not emphasize the standalone applications in the contributions guide. In principle they work without Zeebe technically, but all require Elasticsearch. But then ES needs to be filled with data, which ultimately comes from Zeebe. Let's use the advantage of the monoRepo that we have:
  * I can do a change to camunda/camunda. This change can be in Operate or Zeebe or ... domain.
  * I can build everything (Operate, Zeebe, ...) with one command
  * I can try it out right away (after having started ElasticSearch ;-) ) 

> Are any of these steps planned to be supported in a Docker container (e.g., the camunda.Dockerfile image)?

As part of an camunda release we now have: the single application, and Docker images. The docker images are for the individual components (e.g., Operate, Zeebe). These docker images are used in the distributed deployments (via HELM, our Operator, if you like docker-compose)

**Comment by nloding** [comment]
@MaxTru - that answers almost everything! Three short follow ups:

- if they want to test locally, they will need to have Postgres and/or Elasticsearch installed. I see there is a `config/application.yaml` configuration provided. Can they update those values as needed or do they need to be applied to the running config in a different way?
- do we recommend they use a local installation, containers, or no recommendation for Postgres and Elastic?
- regarding my original question about Docker containers - I more meant for development and debugging. I assume most of the engineers are just running everything locally? Some background: this question came up because a couple community members asked me whether they could develop inside containers rather than maintaining the proper Java environment on their local machine.

**Comment by MaxTru** [comment]
Hi @nloding ,

glad it helps!

> * if they want to test locally, they will need to have Postgres and/or Elasticsearch installed. I see there is a `config/application.yaml` configuration provided. Can they update those values as needed or do they need to be applied to the running config in a different way?

I am mentioning some more points which may be relevant here:
* (I should have mentioned this before, but forgot): Currently the assembly creates `camunda-zeebe-8.6.0-SNAPSHOT.tar.gz`. The naming (`-zeebe`) is misleading, since it is not only Zeebe. This will be renamed to something without `zeebe` in the future. @romansmirnov is there an ETA or issue for this?
* To start locally, we did include a README.txt in the past releases, which explained the following:
```
This is a distribution of Camunda 8.6.0-alpha2

How to run
==========

Prerequisites:
1. Download Elasticsearch 8.9.2 (from https://www.elastic.co/downloads/elasticsearch)
2. For non-production cases, disable Elasticsearch's security packages by setting the xpack.security.* configuration options to false in ELASTICSEARCH_HOME/config/elasticsearch.yml
3. Start Elasticsearch by running ELASTICSEARCH_HOME/bin/elasticsearch (or ELASTICSEARCH_HOME\bin\elasticsearch.bat on Windows)

Start Camunda:
1. Run bin/camunda (or bin\camunda.bat on Windows)
```
I think this instruction is helpful. Alternatively, see what @akeller described in the [docs](https://docs.camunda.io/docs/next/guides/orchestrate-microservices/): 
Bildschirmfoto 2024-07-24 um 08 45 46

* Regarding the `config/application.yaml` -> yes, this can be updated to change the settings. I don't think this is needed though in the default case / it is not needed to do changes to just run locally / try it out. @romansmirnov please correct me if wrong.

> * do we recommend they use a local installation, containers, or no recommendation for Postgres and Elastic?

See above screenshot. For the very easy local development use-case via this repo we just recommend to: 
```
* Download [Elasticsearch 8.9.2](https://www.elastic.co/downloads/past-releases/elasticsearch-8-9-2).
* For non-production cases, disable Elasticsearch's security packages by setting the xpack.security.* configuration options to false in ELASTICSEARCH_HOME/config/elasticsearch.yml.
* Start Elasticsearch by running ELASTICSEARCH_HOME/bin/elasticsearch (or ELASTICSEARCH_HOME\bin\elasticsearch.bat on Windows).
```


> * regarding my original question about Docker containers - I more meant for development and debugging. I assume most of the engineers are just running everything locally? Some background: this question came up because a couple community members asked me whether they could develop inside containers rather than maintaining the proper Java environment on their local machine.

Ah I see. So the question is: as an alternative to the local java development, can one build the docker image right away, if one prefers to work with docker. I don't know the answer. @romansmirnov can you help please?

**Comment by akeller** [comment]
I would recommend linking to setup/install instructions rather than duplicating them here. While installing is critical to solid contributions, I'm concerned about maintenance overhead. That said, this is all very in-flight work and I don't have a great link to offer. For now, you could link to the current plain Java install steps, embedded in the Spring guide [here](https://docs.camunda.io/docs/next/guides/getting-started-java-spring/#step-1-install-camunda-8-self-managed).

If we assume users who contribute can navigate a slimmed-down version of the installation instructions, I recommend linking to the README.txt instructions instead of duplicating them. We had to make them a bit more verbose in the docs for more novice users.

**Review by korthout — COMMENTED** [review]
Thanks @xomiamoore. Sorry for taking my time. I have a few new suggestions. Please have a look

**Comment by romansmirnov** [comment]
> (I should have mentioned this before, but forgot): Currently the assembly creates camunda-zeebe-8.6.0-SNAPSHOT.tar.gz. The naming (-zeebe) is misleading, since it is not only Zeebe. This will be renamed to something without zeebe in the future. @romansmirnov is there an ETA or issue for this?

Currently, there is no ETA, I always kept the priority low therefore. However, this can be of course be done soon-ish, and the effort should be relatively low. I will create and check if @Zelldon or @houssain-barouni could support here in between.

> Regarding the config/application.yaml -> yes, this can be updated to change the settings. I don't think this is needed though in the default case / it is not needed to do changes to just run locally / try it out. @romansmirnov please correct me if wrong.

Depends on how they run it locally. Besides that, to ensure that data gets exported to ES, they would need to configure the ES exporter in `config/application.yaml` (asuming they build the distribution and then use the `camunda.sh` script to start).

> > regarding my original question about Docker containers - I more meant for development and debugging. I assume most of the engineers are just running everything locally? Some background: this question came up because a couple community members asked me whether they could develop inside containers rather than maintaining the proper Java environment on their local machine.

> Ah I see. So the question is: as an alternative to the local java development, can one build the docker image right away, if one prefers to work with docker. I don't know the answer. @romansmirnov can you help please?

To build the docker image, they would need to build by executing

```
docker build \
  --tag camunda/zeebe:local \
  --build-arg DISTBALL='dist/target/camunda-zeebe*.tar.gz' \
  --target app \
  --file ./camunda.Dockerfile
  .
```

However, I don't think I understand @nloding's question correctly

> [...] they could develop inside containers [...]

What means, "inside" containers?

Hope that helps.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -85,53 +80,107 @@ The `main` branch contains the current in-development state of the project. To w
...
+- `dist` 
+- `identity` - component within self-managed Camunda 8 responsible for authentication and authorization
+- `licenses`
+- `monitor` - Monitoring for self-managed Camunda 8
+- `operate` - Monitoring tool for monitoring and troubleshooting processes running in Zeebe
+- `parent` - Parent POM for all Zeebe projects
+- `qa` 
+- `search`
+- `service`
+- `spring-boot-starter-sdk` - official SDK for Spring Boot
```
[the remark this answers is not in the document]
Comment (xomiamoore) [reply CONTRIBUTING.md]: Updated these, and took out the folders that start with `.`

**Comment by xomiamoore** [comment]
Hey team, @nloding and I updated the build and run instructions. We also talked to @akeller and decided to keep the instructions for building from source separate from the installation instructions on the docs site, since they do differ slightly.

Looking forward to your feedback -- I think we're close! 🎉 

cc: @korthout @MaxTru @romansmirnov

**Review by MaxTru — APPROVED** [review]
LGTM!

Please note that I did not verify whether the "branch creation" and "pr creation" guidelines work in practice.

**Comment by MaxTru** [comment]
Hi @xomiamoore ,
I just realized: the CI is failing. I just had a very quick look now and for sure you would need to fix the commit messages to follow the guidelines. Can you please follow-up?

**Comment by xomiamoore** [comment]
@MaxTru  Fixed the commit messages, I see that some checks are still failing. Any hints to fix these?

**Comment by MaxTru** [comment]
> @MaxTru Fixed the commit messages, I see that some checks are still failing. Any hints to fix these?

Hi @xomiamoore ,
your branch was quite outdated (more than 1.000 commits behind). I rebased it for you, should be good now.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -143,53 +81,110 @@ follow the following steps:
...
+- `service` - internal services for Camunda 8
+- `spring-boot-starter-sdk` - official SDK for Spring Boot
+- `tasklist` - graphical and API application to manage user tasks in Zeebe
+- `testing` - testing libraries for processes and process applications
+- `webapps-common` - shared code between the Camunda 8 web apps
+- `zeebe` - the process automation engine powering Camunda 8
+
+### Build
+
+To **quickly** build all components, run the command: `mvn clean install -DskipTests` in the root folder.
```
Comment (korthout) [inline CONTRIBUTING.md:116]: ❓ Why not recommend `mvn clean install -Dquickly` anymore?

The rest of this section implies that building quickly is typically sufficient unless you want a full distribution. However, the current `mvn clean install -DskipTests` provides a full distribution. Note that the `-Dquickly` flag skips all non-essentials goals like assembly of the distribution artifacts. It's useful during development, but not for those that build from source to use the product.

https://github.com/camunda/camunda/blob/f82906ed9e39a6b3305622c26758eb0cf7e314fb/parent/pom.xml#L232-L265

**Comment by MaxTru** [comment]
Given that @xomiamoore and @mary-grace are both out, I am trying to merge this on their behalf. 

@korthout , you had requested changes, can you please confirm this is good now (or comment if not?) Thanks.

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -143,53 +81,110 @@ follow the following steps:
...
+
+> [!NOTE]
+> All Camunda core modules are built and tested with JDK 21. Most modules use language level 21, exceptions are: camunda-client-java, camunda-process-test-java, zeebe-bpmn-model, zeebe-build-tools, zeebe-client-java, zeebe-gateway-protocol zeebe-gateway-protocol-impl, zeebe-protocol, and zeebe-protocol-jackson which use language level 8.
+
+For contributions to Camunda, building quickly is typically sufficient. However, users are recommended to build the full distribution.
+
+To fully build the Camunda distribution and run all tests, run the command: `mvn clean install` in the root folder. This is slightly slower than building quickly but ensures the distribution is assembled completely. The resulting distribution can be found in the folder `dist/target`, i.e.
```
Comment (korthout) [inline CONTRIBUTING.md:123]: ❓ Why do we recommend running all the tests for building the Camunda distribution? Doesn't it make more sense to split it into the following?
- Building quickly for development: `mvn clean install -Dquickly`
- Building the full distribution for usage (you don't want to run all the tests): `mvn clean install -DskipChecks -DskipTests`
- Building and testing the full distribution: `mvn clean install`

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -222,12 +216,9 @@ When this happens and you're still interested in contributing, please feel free
 
 ## Backporting changes
 
-Some changes need to be copied to older versions. We use the
-[backport](https://github.com/zeebe-io/backport-action) Github Action to automate this process.
-Please follow these steps to backport your changes:
+Some changes need to be copied to older versions. We use the [backport](https://github.com/zeebe-io/backport-action) Github Action to automate this process. Please follow these steps to backport your changes:
```
Comment (korthout) [inline CONTRIBUTING.md:219]: 🔧 Shameless brag, the backport-action has been moved (already quite some time ago) and exists under https://github.com/korthout/backport-action. We use this specifically:
```suggestion
Some changes need to be copied to older versions. We use the [backport](https://github.com/korthout/backport-action) Github Action to automate this process. Please follow these steps to backport your changes:
```

**Review by korthout — APPROVED** [review]
@MaxTru I have some final recommendations/questions, but LGTM 👍  

🔧 If you want to get this merged, we can take these comments along separately. I've also provided suggestions that you can commit directly through Github if you want to take them along immediately.

💭 One last thought is that we could add a section specifically to support frontend or backend development. For example, using `-PskipFrontendBuild`. We don't need to add this now.

cc @xomiamoore

File: CONTRIBUTING.md (hunk 13 above)
Comment (MaxTru) [reply CONTRIBUTING.md:116]: I agree. Committing this suggestion. @xomiamoore plesae chime in later if there was another reason for this which we don't see at the moment.

File: CONTRIBUTING.md (hunk 14 above)
Comment (MaxTru) [reply CONTRIBUTING.md]: Makes sense! Thanks. Corrected

**Comment by MaxTru** [comment]
Thanks @korthout ! Followed your guidance and adjusted accordingly. Merging
